flang/lib/Lower/OpenMP.cpp

```diff
@@ -87,19 +88,35 @@ genOMP(Fortran::lower::AbstractConverter &absConv,
     auto &firOpBuilder = absConv.getFirOpBuilder();
     auto currentLocation = absConv.getCurrentLocation();
     auto insertPt = firOpBuilder.saveInsertionPoint();
+
+    // Clauses.
+    mlir::Value numThreads;
+
+    const auto &parallelOpClauseList =
+        std::get<Fortran::parser::OmpClauseList>(blockDirective.t);
+    for (const auto &clause : parallelOpClauseList.v) {
+      if (const auto &numThreadsClause =
+              std::get_if<Fortran::parser::OmpClause::NumThreads>(&clause.u)) {
+        numThreads = firOpBuilder.createIntegerConstant(
+            currentLocation, firOpBuilder.getIntegerType(32),
```

Comment:
Will this always be a 32-bit integer? Or should the type come from somewhere in the parse tree? (Front-end types should be translated from KIND values to MLIR using the KindMapping.)